dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorStoreRecordMapper.cs

```diff
@@ -0,0 +1,299 @@
+        typeof(List<bool>),
+        typeof(string),
+        typeof(int),
+        typeof(long),
+        typeof(double),
+        typeof(float),
+        typeof(bool),
+        typeof(int?),
+        typeof(long?),
+        typeof(double?),
+        typeof(float?),
+        typeof(bool?)
+    ];
+
+    /// <summary>A set of types that vectors on the provided model may have.</summary>
+    /// <remarks>
+    /// While qdrant supports float32 and uint64, the api only supports float64, therefore
+    /// any float32 vectors will be converted to float64 before being sent to qdrant.
+    /// </remarks>
+    private static readonly HashSet<Type> s_supportedVectorTypes =
+    [
+        typeof(ReadOnlyMemory<float>),
+        typeof(ReadOnlyMemory<float>?),
+        typeof(ReadOnlyMemory<double>),
+        typeof(ReadOnlyMemory<double>?)
+    ];
+
+    /// <summary>A list of property info objects that point at the payload properties in the current model, and allows easy reading and writing of these properties.</summary>
+    private readonly List<PropertyInfo> _payloadPropertiesInfo = new();
+
+    /// <summary>A list of property info objects that point at the vector properties in the current model, and allows easy reading and writing of these properties.</summary>
+    private readonly List<PropertyInfo> _vectorPropertiesInfo = new();
+
+    /// <summary>A property info object that points at the key property for the current model, allowing easy reading and writing of this property.</summary>
+    private readonly PropertyInfo _keyPropertyInfo;
+
+    /// <summary>Optional configuration options for this class.</summary>
+    private readonly QdrantVectorStoreRecordMapperOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QdrantVectorStoreRecordMapper{TDataModel}"/> class.
+    /// </summary>
+    /// <param name="options">Optional options to use when doing the model conversion.</param>
+    public QdrantVectorStoreRecordMapper(QdrantVectorStoreRecordMapperOptions? options)
+    {
+        this._options = options ?? new QdrantVectorStoreRecordMapperOptions();
+
+        // Enumerate public properties using configuration or attributes.
+        (PropertyInfo keyProperty, List<PropertyInfo> dataProperties, List<PropertyInfo> vectorProperties) properties;
+        if (this._options.VectorStoreRecordDefinition is not null)
+        {
+            properties = VectorStoreRecordPropertyReader.FindProperties(typeof(TRecord), this._options.VectorStoreRecordDefinition, supportsMultipleVectors: this._options.HasNamedVectors);
+        }
+        else
+        {
+            properties = VectorStoreRecordPropertyReader.FindProperties(typeof(TRecord), supportsMultipleVectors: this._options.HasNamedVectors);
+        }
+
+        // Validate property types and store for later use.
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes([properties.keyProperty], s_supportedKeyTypes, "Key");
```

Comment:
nit: I just noticed that `VectorStoreRecordPropertyReader` class is located in `Schema` folder of internal utilities, and it looks like this folder is related to JSON schema. I'm wondering if we should put `VectorStoreRecordPropertyReader` class in separate folder.

https://github.com/microsoft/semantic-kernel/tree/feature-memory-service/dotnet/src/InternalUtilities/src/Schema
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/7ee4b566-7ce9-4d5e-a667-c7c6548078be)

Reply:
No worries, I hadn't seen the readme, so didn't realize there was a common plan to the schema ones.  I'll move it to a new data folder.